Assets/Scripts/PlayerMovement/CatMovement.cs

```diff
@@ -388,10 +390,19 @@ IEnumerator SetObject()
         while (map == null)
         {
             map = GameObject.Find("Map");
-            //blockManager = FindObjectOfType<BlockManager>();
             yield return new WaitForEndOfFrame();
         }
         blockManager = map.GetComponentInChildren<BlockManager>();
+        try
+        {
+            transform.position = map.GetComponentInChildren<StartingBlock>().GetPos();
+        }
+        catch
+        {
+            Debug.Log("starting block not made");
+            transform.position = new Vector3(0, 0, 0);
+        }
```

Comment:
Honestly not a fan of try catch statements. They catch all errors, and they tend to nest fast if you want to catch specific errors. Imo they're best used only when a certain line of code can cause a multitude of problems and it's impossible to predict/handle every one (e.g. database queres, API requests, etc.)

In this case you do know what the problem is. If you want to add a guard for the addition of the starting block, I'd say just be explicit and check if `map.GetComponentInChildren<StartingBlock>()` is defined before setting `transform.position` to its `GetPos` function.